hangman.rb

```diff
@@ -0,0 +1,18 @@
+# Hardcode solutionds
+def hangman(word, guess)
+  word.gsub(/[^#{guess.join('')}]/, "_")
+end
+
+puts hangman('hangman', ['h', 'g'])
+
+# Interactive solution
+def hangman(word)
+  guess = []
+  until guess.last == ""
+    print "Guess a letter in the word, or hit enter to see the result: "
+    guess << gets.chomp
+  end
+  word.gsub(/[^#{guess.join('')}]/, "_")
+end
```

Comment:
This didn't work how I would have expected as a player. After making a guess there isn't any feedback for whether the guess was correct or not. If you move line 15 (`word.gsub(/[^#{guess.join('')}]/, "_")`) into the `until` block and add a `puts` at the beginning of the line then it will print out the current state of the "hangman" word after each guess.

Reply:
Has revised it to make it full-blown hangman game
```ruby
def hangman(word)
  guesses = ''
  until guesses == ' '
    print 'Guess a letter in the word, or press enter to quit: '
    guess = gets.chomp
    if guess.empty?
      guesses = ' '
    elsif guess.match?(/[^A_Za-z]/)
      puts "\nYou must enter a letter. Try again.\n\n"
    else
      guesses = guesses.include?(guess) ? guesses : guesses + guess
      if word.gsub(/[#{guesses}]/, '') == ''
        puts "\nYou won: #{word}"
        guesses = ' '
      else
        puts "\n#{word.include?(guess) ? 'Hit:' : 'Miss:'} #{word.gsub(/[^#{guesses}]/, '_')}\n\n"
      end
    end
  end
end
```